src/main/java/picard/analysis/SinglePassSamProgram.java

```diff
@@ -56,7 +56,7 @@
     @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input SAM or BAM file.")
     public File INPUT;
 
-    @Argument(shortName = "O", doc = "File to write the output to.")
+    @Argument(shortName = "O", doc = "File to write the output to.", optional = true)
```

Comment:
why is this optional?

Reply:
`CollectRrbsMetrics` didn't have `OUTPUT` argument (only `METRICS_FILE_PREFIX`). But now it extends from `SinglePassSamProgram` and you can't run the metrics without **mandatory  argument** `OUTPUT`. 

So `OUTPUT` was made optional in order to be able to run `CollectRrbsMetrics` without it.

Also we can use that excessive `OUTPUT` argument with two more to specify output files explicitly. It can be useful and it is a way to use this unnecessary argument.

Reply:
ok.

Reply:
thinking about this more, this isn't "OK". this changes the behavior of all the other programs that extend from SinglePassSamProgram.

can you find a solution that doens't change the behavior/API for the other programs?

Reply:
I owuld almost prefer to simply rename METRICS_FILE_PREFIX -> "OUTPUT" and to do away with the specific metric/chart/summary output variables....the only problem is that it would create an API change...

@cnorman, perhaps you can suggest solutions for this...given your knowledge of the parsing system.

Reply:
I suggest that you implement it like CollectGcBiasMetrics is implemented, i.e. OUTPUT is for the detailed_metrics, SUMMARY_OUTPUT is for summary_metrics and CHART_OUTPUT for the pdf. You can use METRICS_FILE_PREFIX as Mutex, or remove it and change the API for this program (but not for all of the Single-Pass-Sam programs)